File: lab10/Task/Delegations/Population.cs
public class Population
{
    private readonly Person[] _people;

    public Population(int size)
    {
        _people = new Person[size];
    }

    public void SetPerson(int index, Person person)
    {
        lock (this)
        {
            _people[index] = person;
        }
    }

    public Person GetPerson(int id)
    {
        return (id >= 0 && id < _people.Length) ? _people[id] : null;
    }

    public void UpdatePerson(int id, bool isInfected, int infectionDays, bool hasRecovered)
    {
        lock (this)
        {
            if (id >= 0 && id < _people.Length)
            {
                _people[id].IsInfected = isInfected;
                _people[id].InfectionDays = infectionDays;
                _people[id].HasRecovered = hasRecovered;
            }
        }
    }

    public int GetSize()
    {
        return _people.Length;
    }
}

File: Lab4/maven/src/main/java/org/example/Population.java
package org.example;


public class Population {
    private final Person[] people;

    public Population(int size) {
        this.people = new Person[size];
    }

    public void setPerson(int index, Person person) {
        synchronized (this) {
            people[index] = person;
        }
    }

    // Thread-safe method to read a person
    public Person getPerson(int id) {
        if (id >= 0 && id < people.length) {
            return people[id];
        }
        return null;
    }

    // Thread-safe method to update a person
    public void updatePerson(int id, boolean isInfected, int infectionDays, boolean hasRecovered) {
        synchronized (this) {
            if (id >= 0 && id < people.length) {
                people[id].isInfected = isInfected;
                people[id].infectionDays = infectionDays;
                people[id].hasRecovered = hasRecovered;
            }
        }
    }

    public int getSize() {
        return people.length;
    }
}

